### ISSUE TITLE
webpack compatibility for serverless framework -> lambda

### ISSUE THREAD

**Issue by wyvern8** [octokit/octokit.js#774]
Hi @gr2m , i have made some progress on the issue reported here https://github.com/octokit/rest.js/commit/d191dfc7096df038d767f2000d11caa81b719375#commitcomment-27295686 

There are two places causing issues with webpack.

### 1. is-array-buffer module.
webpack does not seem to like 'export default', although in this link they could use it by hoisting function first.  https://github.com/webpack/webpack/issues/2399

due to the package.json module field of is-array-buffer, the file picked up is **is-array-buffer.esm.js**.

this file uses es6: 
```
export default isArrayBuffer;
```
.. however if i change it to contain the same as 'is-array-buffer.common.js':
```
module.exports = isArrayBuffer;
```
..it works with webpack.  given this module is about 5 lines of code, it might be simplest to incorporate n octokit with attribution?  https://github.com/fengyuanchen/is-array-buffer/blob/master/dist/is-array-buffer.common.js 

### 2. webpack does not like require calls containing expressions.
https://github.com/webpack/webpack/issues/2413

in octokit get-request-agent.js

.. changing this:
```
const Agent = require(options.protocol || DEFAULTS.protocol).Agent
```
..to this..
```
let Agent;
if (options.protocol && options.protocol === 'http') {
	Agent = require('http').Agent;
} else {
	Agent = require('https').Agent
}	    
```
..works, but you might want to handle it differently.

**So, would you like a pull request with the above 2 items, or prefer to handle this yourself?**

i'm pretty keen to continue using webpack for our lambdas, as it bring size down from 64Mb to 3Mb each. fyi @Neko-Design

**Comment by gr2m** [comment]
Thanks @wyvern8, could you send two separate pull requests please?

Just a quick tip, you past a permalink URL into a comment, it will show the actual code and referencing the right file, so it’s easier for others to understand where to code is you are referencing

![image](https://user-images.githubusercontent.com/39992/36867015-bfb29be8-1d48-11e8-90e9-8f8474e6a777.png)

e.g. `https://github.com/octokit/rest.js/blob/57c20ed25ca6da640571db0e9cea9dfab11597b3/lib/get-request-agent.js#L20` will result in:

https://github.com/octokit/rest.js/blob/57c20ed25ca6da640571db0e9cea9dfab11597b3/lib/get-request-agent.js#L20

Regarding `is-array-buffer module.`, putting the module into our code works for me. We will probably use es modules for octokit itself in future for better tree shaking support (mostly to reduce bundle size for browser usage), I guess at this point we should be able to use the module again

**Comment by wyvern8** [comment]
ok thanks @gr2m - will do.

**Comment by wyvern8** [comment]
^ here you go @gr2m

**Comment by gr2m** [comment]
Great thanks, just left a few style-related comments. If you are busy, I can finish it up for you. I want to get your PRs merged and released before #764, even though it will most likely remove `is-array-buffer`, just in case the migration to node-fetch brings any new trouble that I’m not aware of yet :)

**Comment by wyvern8** [comment]
ok - all upgraded and tested fine in https://github.com/wyvern8/github-task-manager/pull/61 so closing this one - thanks for your help with this @gr2m - much appreciated.
